sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer_client.py

```diff
@@ -0,0 +1,184 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import logging
+from typing import Any, Union, TYPE_CHECKING
+from ._common import EventPosition,\
+    EventHubSharedKeyCredential, EventHubSASTokenCredential
+from ._eventprocessor.event_processor import EventProcessor
+from ._client import EventHubClient
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential  # type: ignore
+
+log = logging.getLogger(__name__)
+
+
+class EventHubConsumerClient(EventHubClient):
+    """Represents an AMQP connection to an EventHub and receives event data from it.
```

Comment:
Class level and init level doc doesn't play well together, I would suggest to merge this string inside the `__init__`